### ISSUE TITLE
First run with Puppet in strict mode fails due to missing fact value

### ISSUE THREAD

**Issue by ross-w** [duritong/puppet-shorewall#18]
When we use this module on a fresh install (PE 2017.2.3, strict variables), the puppet run fails:
```
Error: Could not retrieve catalog from remote server: Error 500 on SERVER: Server Error: Evaluation Error: Error while evaluating a Resource Statement, Evaluation Error: Unknown variable: 'shorewall_version'. at /etc/puppetlabs/code/environments/production/modules/shorewall/manifests/interface.pp:22:17
```

This is due to trying to access `$shorewall_version` when Shorewall isn't installed yet, so no fact value. This is fixed by manually installing shorewall, but this can't happen in Puppet as the run aborts with the above error during compilation.

Could the module check for the existence of this fact before proceeding to use it?

**Comment by duritong** [comment]
Yeah, that's likely the problem I'm still wondering what the best strategy is to deal with `strict = true` in manifests. Anyone knows a best practice?
